scripts/globals/spells/bluemagic/pinecone_bomb.lua

```diff
@@ -0,0 +1,65 @@
+-----------------------------------------
+-- Spell: Pinecone Bomb
+-- Additional effect: Sleep. Duration of effect varies with TP
+-- Spell cost: 48 MP
+-- Monster Type: Plantoids
+-- Spell Type: Physical (Piercing)
+-- Blue Magic Points: 2
+-- Stat Bonus: STR+1
+-- Level: 36
+-- Casting Time: 2.5 seconds
+-- Recast Time: 26.5 seconds
+-- Skillchain Element(s): Fire (can open Scission or Fusion and can close Liquefaction)
+-- Combos: None
+-----------------------------------------
+require("scripts/globals/bluemagic")
+require("scripts/globals/status")
+require("scripts/globals/magic")
+-----------------------------------------
+
+function onMagicCastingCheck(caster,target,spell)
+    return 0
+end
+
+function onSpellCast(caster,target,spell)
+    local params = {}
+    -- This data should match information on http://wiki.ffxiclopedia.org/wiki/Calculating_Blue_Magic_Damage
+    params.tpmod = TPMOD_CRITICAL
+    params.dmgtype = DMGTYPE_PIERCE
+    params.scattr = SC_LIQUEFACTION
+    params.numhits = 1
+    params.multiplier = 1.25
+    params.tp150 = 1.25
+    params.tp300 = 1.25
+    params.azuretp = 1.25
+    params.duppercap = 37
+    params.str_wsc = 0.20
+    params.dex_wsc = 0.0
+    params.vit_wsc = 0.0
+    params.agi_wsc = 0.20
+    params.int_wsc = 0.0
+    params.mnd_wsc = 0.0
+    params.chr_wsc = 0.0
+
+    local damage = BluePhysicalSpell(caster, target, spell, params)
+    damage = BlueFinalAdjustments(caster, target, spell, damage, params)
+
+    -- After damage is applied (which would have woken the target up from a
+    -- preexisting sleep, if necesesary), apply the sleep effect for this spell.
+    if (damage > 0) then
+        local sleepParams = {}
+        sleepParams.attribute = tpz.mod.INT
+        sleepParams.bonus = 0
+        sleepParams.effect = tpz.effect.SLEEP_II
+        sleepParams.skillType = tpz.skill.BLUE_MAGIC
+        local resist = applyResistanceEffect(caster, target, spell, sleepParams)
+        if (resist > 0.5) then -- Apply the sleep
+            local power = 2
+            local tick = 0
+            local duration = 90 * resist
```

Comment:
This was my first time adding a BLU spell, so I'm open to any improvements.

I just copied how sleep was applied in other spells, so it may not be 100% accurate. If you have a reference for how this should be applied, please point me in the right direciton!

Reply:
What we'll probably have to do is find some information - or get a capture - on the basic duration of the effect. Even if we can't fully pick apart how TP affects the duration for this (or other) Blue spells, we can at least apply the bare minimum duration for now and add a note in the script (and create a new Issue) that it's still missing duration bonuses from TP.

Reply:
Sounds good.

Should I wait on someone to do a "capture"? Or should I go forward with a TODO note and reference a github issue to update the TP modification of the sleep duration?

Reply:
We have some captures. I'll check them out and see what I can determine. Then hopefully someone more experienced can double check me.